### PR TITLE
Execute query via cli

### PR DESCRIPTION
**Pull request by 0xkafkaa**
#7 Execute query via CLI 
- [x] Added the functionality to execute query via CLI.
- Right now the command works when running the docker container. If the functionality seems OK, will proceed to integrate the docker comand in the shell script. 
Check out this [demo](https://asciinema.org/a/zbZVMNuAdTiV5ylOWFKTf2eUx).

Regards,
0xkafkaa.

### CODE REVIEW

**Comment by Cereal84** [comment]
> #7 Execute query via CLI 
> - [x] Added the functionality to execute query via CLI.
> - Right now the command works when running the docker container. If the functionality seems OK, will proceed to integrate the docker comand in the shell script. 
> Check out this [demo](https://asciinema.org/a/zbZVMNuAdTiV5ylOWFKTf2eUx).
> 
> Regards,
> 0xkafkaa.

It will be better to have a -f FILE -q QUERY instead of use -d FILE.
-d Is used to specify a  directory.

**Comment by 0xkafkaa** [comment]
- [x] It will be better to have a -f FILE -q QUERY instead of use -d FILE.
-d Is used to specify a directory.

Let me know your comments.
Screenshot_2024-09-26_14-56-13

File: anadi/__main__.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 parser = argparse.ArgumentParser(prog="AnadiCSV", description="Handle CVS file using SQL language")
 parser.add_argument("-d", "--dir", type=str, help="CSV file dir", required=True)
```
Comment (Cereal84) [inline anadi/__main__.py]: this one is not required anymore BUT we need to check if it has a value if -q and -f are not valueated

File: anadi/__main__.py [hunk 2]
```diff
@@ -19,7 +20,7 @@ def run():
         app.init(args.dir, os.path.abspath(os.path.expanduser(ANADI_CONF_FILE)))
         if args.query:
             csvdb = CSVDB()
-            csvdb.load(args.dir,SettingsDB())
+            csvdb.load(args.file,SettingsDB())
             result = csvdb.raw_sql(args.query)
             print(result)
```
Comment (Cereal84) [inline anadi/__main__.py]: if *-d* has no value we need to chcek if both *-q* and *-f* have value.

**Review by Cereal84 — CHANGES_REQUESTED** [review]

File: anadi/__main__.py (hunk 1 above)
Comment (0xkafkaa) [reply anadi/__main__.py]: Now either the dir or the file and query need to be passed.

File: anadi/__main__.py (hunk 2 above)
Comment (0xkafkaa) [reply anadi/__main__.py]: Now either the dir or the file and query need to be passed. Hence this as well taken care

**Comment by 0xkafkaa** [comment]
Now, if the d flag is passed it opens the editor and if both f and q flags are passed,  the result is printed in the cli. If any invalid agrs are passsed corresponding errors are raised.

File: anadi/__main__.py [hunk 3]
```diff
@@ -7,24 +7,29 @@
...
+            app.init(args.dir, os.path.abspath(os.path.expanduser(ANADI_CONF_FILE)))
+            app.run()
+        elif args.file and args.query:
             csvdb = CSVDB()
             csvdb.load(args.file,SettingsDB())
             result = csvdb.raw_sql(args.query)
             print(result)
-        else:
-            app.run()
+
```
Comment (Cereal84) [inline anadi/__main__.py]: if *args.dir* and *args.query* are set but *args.file* not we execute *args.dir* anyway and not error should be shown.

So or the condition is improved (but if we add another option that check became bigger) or we decide to remove this check and error and let the -d option win other the others.

Any idea ?

Comment (0xkafkaa) [reply anadi/__main__.py]: Initially, I let the args.dir win. If args.dir  and args.file are set not args.query I opened the editor. But the user experience doesn't seem good.
Its either directory in the editor or the file name with the query.
We could make this clear in the bash script help flag for better usability. 

Works??

**Comment by Cereal84** [comment]
@0xkafkaa please resolve the conflicts

**Review by Cereal84 — APPROVED** [review]

**Review by Cereal84 — APPROVED** [review]

**Comment by Cereal84** [comment]
thak you @0xkafkaa

**Pull request merged**